api/python/quilt3/data_transfer.py

```diff
@@ -512,6 +543,9 @@ def done_callback(value):
 
     assert all(results)
 
+    # Close the executor
+    _close_executor(executor)
```

Comment:
Should be done before the `assert`. (Though bringing back `with` will make this go away, anyway.)